File: antigravity-ide/brain/1b0ac5d1-a519-4a62-bbb8-1382c9f0c55a/scratch/swap_nav.js
const fs = require('fs');

// Read owner dashboard
const ownerHtml = fs.readFileSync('c:/Users/VICTUS/.gemini/antigravity-ide/scratch/petcare-app/owner-dashboard.html', 'utf8');

// Extract appSidebar
const sidebarMatch = ownerHtml.match(/(<nav id="appSidebar"[\s\S]*?<\/nav>)/);
if (!sidebarMatch) {
    console.error("Error: Could not find appSidebar in owner-dashboard.html");
    process.exit(1);
}
const ownerSidebar = sidebarMatch[1];

// Extract mobile nav
const mobileNavMatch = ownerHtml.match(/(<nav class="bg-surface border-t border-outline-variant md:hidden fixed bottom-0 left-0 w-full z-50 flex justify-around items-center px-4 py-2">[\s\S]*?<\/nav>)/);
if (!mobileNavMatch) {
    console.error("Error: Could not find mobile nav in owner-dashboard.html");
    process.exit(1);
}
const ownerMobileNav = mobileNavMatch[1];

// Read vet-profile
let vetHtml = fs.readFileSync('c:/Users/VICTUS/.gemini/antigravity-ide/scratch/petcare-app/vet-profile.html', 'utf8');

// Replace appSidebar in vet-profile
vetHtml = vetHtml.replace(/<nav id="appSidebar"[\s\S]*?<\/nav>/, ownerSidebar);

// Ensure mobile nav exists in vet-profile, if it doesn't, append before </body>
if (!vetHtml.includes('<nav class="bg-surface border-t border-outline-variant md:hidden fixed bottom-0')) {
    vetHtml = vetHtml.replace('</body>', ownerMobileNav + '\n</body>');
} else {
    vetHtml = vetHtml.replace(/<nav class="bg-surface border-t border-outline-variant md:hidden fixed bottom-0[\s\S]*?<\/nav>/, ownerMobileNav);
}

// Replace the title to make it clear it's the Owner's view of the vet profile
vetHtml = vetHtml.replace('<title>Dr. Ananya Sharma - Doctor Profile | PetCare</title>', '<title>Veterinarian Profile | PetCare</title>');

fs.writeFileSync('c:/Users/VICTUS/.gemini/antigravity-ide/scratch/petcare-app/vet-profile.html', vetHtml, 'utf8');

console.log("Successfully updated vet-profile.html with Owner Navigation!");

File: antigravity-ide/brain/1b0ac5d1-a519-4a62-bbb8-1382c9f0c55a/scratch/swap_nav.py
import re

# Read owner dashboard
with open('c:/Users/VICTUS/.gemini/antigravity-ide/scratch/petcare-app/owner-dashboard.html', 'r', encoding='utf-8') as f:
    owner_html = f.read()

# Extract appSidebar
sidebar_match = re.search(r'(<nav id="appSidebar".*?</nav>)', owner_html, re.DOTALL)
if not sidebar_match:
    print("Error: Could not find appSidebar in owner-dashboard.html")
    exit(1)
owner_sidebar = sidebar_match.group(1)

# Extract mobile nav
mobile_nav_match = re.search(r'(<nav class="[^"]*md:hidden fixed bottom-0[^>]*>.*?</nav>)', owner_html, re.DOTALL)
if not mobile_nav_match:
    # Alternative regex for mobile nav
    mobile_nav_match = re.search(r'(<nav class="bg-surface border-t border-outline-variant md:hidden fixed bottom-0 left-0 w-full z-50 flex justify-around items-center px-4 py-2">.*?</nav>)', owner_html, re.DOTALL)
    if not mobile_nav_match:
        print("Error: Could not find mobile nav in owner-dashboard.html")
        exit(1)
owner_mobile_nav = mobile_nav_match.group(1)

# Read vet-profile
with open('c:/Users/VICTUS/.gemini/antigravity-ide/scratch/petcare-app/vet-profile.html', 'r', encoding='utf-8') as f:
    vet_html = f.read()

# Replace appSidebar in vet-profile
vet_html_new, num_subs = re.subn(r'<nav id="appSidebar".*?</nav>', owner_sidebar, vet_html, count=1, flags=re.DOTALL)
if num_subs == 0:
    print("Error: Could not replace appSidebar in vet-profile.html")
    exit(1)

# Ensure mobile nav exists in vet-profile, if it doesn't, append before <script>
if '<nav class="bg-surface border-t border-outline-variant md:hidden fixed bottom-0' in vet_html_new:
    vet_html_new, num_subs = re.subn(r'<nav class="bg-surface border-t border-outline-variant md:hidden fixed bottom-0.*?</nav>', owner_mobile_nav, vet_html_new, count=1, flags=re.DOTALL)
else:
    # insert before script tags
    vet_html_new = vet_html_new.replace('<script>', owner_mobile_nav + '\n    <script>', 1)
    # wait, script might be something else
    # fallback to before </body>
    if owner_mobile_nav not in vet_html_new:
        vet_html_new = vet_html_new.replace('</body>', owner_mobile_nav + '\n</body>')

# Replace the title to make it clear it's the Owner's view of the vet profile
vet_html_new = vet_html_new.replace('<title>Dr. Ananya Sharma - Doctor Profile | PetCare</title>', '<title>Veterinarian Profile | PetCare</title>')

with open('c:/Users/VICTUS/.gemini/antigravity-ide/scratch/petcare-app/vet-profile.html', 'w', encoding='utf-8') as f:
    f.write(vet_html_new)

print("Successfully updated vet-profile.html with Owner Navigation!")
